test/util/range.cpp

```diff
@@ -213,6 +213,21 @@ void test_pointers()
     BOOST_CHECK(bgr::at(r2, 5) == 5);
 }
 
+void test_const_pointers()
```

Comment:
We have had this discussion in the past. I do not remember if it was internal (between the main BG developers only or not), and the outcome was to stick with header-only also in tests.

Whether it is weird or not, it is really up to your point-of-view. I can see advantages in both cases. From the library's point-of-view, staying with header-only means one less requirement/dependency.